### ISSUE TITLE
Property 'mainClass' not set in 'secondaryLauncher' block

### ISSUE THREAD

**Issue by HGuillemet** [beryx/badass-jlink-plugin#213]
When defining a secondary launcher with Kotlin DSL:
```Kotlin
jlink {
    secondaryLauncher {
        name = "my-additional-app"
        mainClass.set("org.example.MyAdditionalApp")
        args = listOf("--user", "emma")
    }
}
```
Gradle complains with:
```
Property 'mainClass' not set in 'secondaryLauncher' block
```
Is there something I'm doing wrong ?

BTW, the example in the documention for Kotlin DSL reads:
```
        mainClass = "org.example.MyAdditionalApp"
```
which also triggers an error:
```
Type mismatch: inferred type is String but Property<String!>! was expected
```

**Comment by airsquared** [comment]
Same issue with Groovy DSL.

```
Caused by: org.gradle.api.GradleException: Property 'mainClass' not set in 'secondaryLauncher' block
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	at org.beryx.jlink.data.SecondaryLauncherData$_check_closure1.doCall(SecondaryLauncherData.groovy:39)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at org.beryx.jlink.data.SecondaryLauncherData.check(SecondaryLauncherData.groovy:38)
	at org.beryx.jlink.data.JlinkPluginExtension.secondaryLauncher(JlinkPluginExtension.groovy:182)
	at org.beryx.jlink.data.JlinkPluginExtension_Decorated.secondaryLauncher(Unknown Source)
```

**Comment by airsquared** [comment]
Managed to get past this by also specifying `moduleName` but then I run into https://github.com/adoptium/adoptium-support/issues/769.

**Comment by airsquared** [comment]
Fixed in v3.0.0.
